### DISCUSSION TITLE
I dont't understand how to select the python version of the wheel I want to publish

### DISCUSSION THREAD

**Discussion by julienfr112** [PyO3/maturin#1125] [Q&A] [answered]
When I publish from the docker image ghcr.io/pyo3/maturin (from linux) i get a python 3.7 manylinuxversion of the wheel, that work great on python 3.7, but for the other version of python on linux. pip install fall back to compiling the module, which is quite a pain for users that dont want to have rust installed localy.

Idealy, I would build for each combination, with cross compiling from rust:

x86 windows  python 3.6, 3.7 ...
x86 linux x86 3.7, 3.8 .....

I did not find in the doc how to do that. Maybe there is a docker image for each ?

**Comment by messense** [comment]
What command were you using to publish wheels?

**Reply by julienfr112** [reply]
Mathurin publish in the docker image. My project use pyo3.

**Reply by messense** [reply]
Have you tried `maturin publish --find-interpreter`?

**Comment by julienfr112** [comment, marked as answer]
My problem was solved by looking at "abi3". I did not now about this, so did not now what to search for ...

just add abi3 in the pyo3 features. Works amazingly well, great job guys !
